docs/our-approach-to-data.md

```diff
@@ -398,10 +398,50 @@ values are persisted we will not be able to reliably tell when the application i
 
 If persisting state causes application errors, opting out of persistence is straightforward: in the `createReducer` util
 provide only the default state value as a first param and don't provide a schema as a third param. Behind the scenes 
-data is never going to be persisted and always regenerated with default value. In this example, it happens to be `'CHECKING'`
+data is never going to be persisted and is always regenerated with default value. In this example, it happens to be `'CHECKING'`
 ```javascript
 export const connectionState = createReducer( 'CHECKING', {
 	[CONNECTION_LOST]: () => 'OFFLINE',
 	[CONNECTION_RESTORED]: () => 'ONLINE'
 } );
 ```
+
+### Opt-in to Persistence ( [#13359](https://github.com/Automattic/wp-calypso/pull/13359) )
+
+Currently reducers are persisted by default if no handlers are given for `SERIALIZE` and `DESERIALIZE`. This is a major 
+problem since many people may not realize that this is happening, and we can run into the data shape errors as 
+noted above.
+
+In the **future** we can opt-in to persistence by adding a schema as a property on the reducer. We do this by combining 
+all of our reducers using `combineReducersWithPersistence` at every level of the tree instead of [combineReducers](http://redux.js.org/docs/api/combineReducers.html).
+Each reducer is then wrapped with `withSchemaValidation` which returns a wrapped reducer that validates on `DESERIALZE` 
+if a schema is present and returns initial state on both `SERIALIZE` and `DESERIALZE` if a schema is not present.
+
+To opt-out of persistence we just combine the plain reducers.
+```javascript
+return combineReducersWithPersistence( {
+    age,
+    height,
+} );
+```
+
+To persist, we add the schema as a property on the reducer:
+```javascript
+age.schema = ageSchema;
+return combineReducersWithPersistence( {
+    age,
+    height,
+} );
+```
+
+For a reducer that has custom handlers (needs to perform transforms), we assume the reducer is checking the schema already, 
+on `DESERIALIZE` so all we need to do is set a boolean bit on the reducer, to ensure that we don't return initial state
+incorrectly from the default handling provided by `withSchemaValidation`.
+```javascript
+date.hasCustomPersistence = true;
+return combineReducersWithPersistence( {
+    age,
+    height,
+    date,
+} );
```

Comment:
Right we only have `SERIALIZE` and `DESERIALIZE` because our tree might contain things that we won't be able to persist without a transform. Validation of the data-shape was just easy to hook into `DESERIALIZE`. 

If our tree was guaranteed to be a plain object, I think we might have an event like `VALIDATE` instead that we call.